apps/backend/src/services/stravaService.ts

```diff
@@ -27,20 +25,18 @@ export const checkStravaConfig = () => {
 };
 
 export const uploadFileToStrava = async (
-  upload: TcxFileUpload,
+  fileContent: string,
+  activityName: string,
   accessToken: string
 ): Promise<Status<StravaUpload, string>> => {
-  const fileStream = Readable.from(upload.tcxFile);
-
   const formData = new FormData.default();
-  formData.append('file', fileStream, {
+  formData.append('file', fileContent, {
     filename: 'dundring.tcx',
     contentType: 'application/xml',
   });
   formData.append('data_type', 'tcx');
 
-  const name = upload.name || 'Dundring.com workout';
-  formData.append('name', name);
+  formData.append('name', activityName);
```

Comment:
No idea what how this works, but I had a quick look at the usage of `FormData`, since I found it kinda surprising that we needed to include a dependency for it – but I guess the dependency made it easier to manage 🤷

Reply:
Seems like this snippet does the same, and it makes it possible to omit the dependency and it makes it possible to omit the cast to `any` in `authPostForm`?

```typescript
export const uploadActivity = async (
  token: string,
  file: string,
  name: string | null
) => {
  const formData = new FormData();
  formData.append(
    'file',
    new Blob([file], { type: 'application/xml' }),
    'dundring.tcx'
  );

  const url = new URL(`${httpUrl}/me/upload`);
  if (name) {
    url.searchParams.append('name', name);
  }

  return authPostForm<ApiResponseBody<StravaUpload>>(url, token, formData);
};
````

Reply:
Fixed. Works 🚀